sklearn/feature_selection/tests/test_feature_select.py

```diff
@@ -169,6 +169,135 @@ def test_f_regression_center():
     assert_almost_equal(F2[0], 0.232558139)  # value from statsmodels OLS
 
 
+@pytest.mark.parametrize(
+    "X, y, expected_corr_coef, force_finite",
+    [
+        (
+            # A feature in X is constant - forcing finite
+            np.array([[2, 1], [2, 0], [2, 10], [2, 4]]),
+            np.array([0, 1, 1, 0]),
+            np.array([0.0, 0.32075]),
+            True,
+        ),
+        (
+            # The target y is constant - forcing finite
+            np.array([[5, 1], [3, 0], [2, 10], [8, 4]]),
+            np.array([0, 0, 0, 0]),
+            np.array([0.0, 0.0]),
+            True,
+        ),
+        (
+            # A feature in X is constant - not forcing finite
+            np.array([[2, 1], [2, 0], [2, 10], [2, 4]]),
+            np.array([0, 1, 1, 0]),
+            np.array([np.nan, 0.32075]),
+            False,
+        ),
+        (
+            # The target y is constant - not forcing finite
+            np.array([[5, 1], [3, 0], [2, 10], [8, 4]]),
+            np.array([0, 0, 0, 0]),
+            np.array([np.nan, np.nan]),
+            False,
+        ),
+    ],
+)
+def test_r_regression_force_finite(X, y, expected_corr_coef, force_finite):
+    """Check the behaviour of `force_finite` for some corner cases with `r_regression`.
+
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/15672
+    """
+    with pytest.warns(None) as records:
+        corr_coef = r_regression(X, y, force_finite=force_finite)
+    assert not [str(w.message) for w in records]
+    np.testing.assert_array_almost_equal(corr_coef, expected_corr_coef)
```

Comment:
Nit: NumPy recommends using `assert_allclose`  [when possible](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_almost_equal.html). We would need to update `expected_corr_coef` to have more significant figures.

Reply:
Yep but it will not work with `np.inf`